### DISCUSSION TITLE
Allow specifying workflow run when creating a checkrun from a GitHub workflow

### DISCUSSION THREAD

**Discussion by callada** [community/community#14891] [Actions]
There is a super old issue that affects manually created check runs, see: https://github.community/t/github-actions-status-checks-created-on-incorrect-check-suite-id/16685, https://github.community/t/specify-check-suite-when-creating-a-checkrun/118380

`POST /repos/{owner}/{repo}/check-runs` adds checks to the incorrect check suite when called from GitHub actions. This creates quite a confusion in the status of our actions.

It would be great if we could specify workflow run id (or check suite id) when creating a check run.

**Comment by jimmylai** [comment]
This issue was reported in Oct. 2019 and we're still waiting for a fix as of May 2022.
https://github.community/t/github-actions-status-checks-created-on-incorrect-check-suite-id/16685
https://github.community/t/specify-check-suite-when-creating-a-checkrun/118380
https://github.community/t/associate-a-check-run-with-a-workflow/122540

**Comment by dclayton-godaddy** [comment]
Any word on this? When a developer views a workflow and does not see the checks that should be associated with it, it slows them down when they have to hunt for the results. Any way we can get this prioritized? Is there a tracking ticket?

**Comment by chrispat** [comment]
It would be helpful if you could provide some scenarios that you are trying to accomplish by creating your own check_runs.  With the job summaries feature and directly creating annotations on the current job via the issue workflow command, I don't have any good additional ones in mind.

**Reply by tkinz27** [reply]
So I configured mikepenz/action-junit-report and this issue comment summarizes at least one additional missing use case (failed tests in the checks UI itself). This action is why I started watching this discussion. 

https://github.com/mikepenz/action-junit-report/issues/40

**Reply by sheck** [reply]
@chrispat I think the biggest thing missing from this alternative is the ability to mark workflow conclusion status as "neutral".

We migrated some check run creation in linting actions to use workflow output commands (and made the actions much simpler!), but we had to do a major release as this was a significant change (a negative one, I'd say) in end-user experienced behavior.

**Comment by KD-7** [comment]
This issue is particularly prevelant when using workflow_dispatch.
When manually running a workflow, where the report is published as a check run annotation, the report gets associated with **ANY** existing workflow run. So if you have 10 workflow runs (completed or incomplete), the report could be published to **ANY** of them. This is a huge flaw in the current API.

**Comment by dd-jonas** [comment]
Also running into this issue, still no updates on this?

**Comment by jamacku** [comment]
I'm also hitting this issue. Do you have any updates or possible workarounds? Thanks

My use-case: I'm running some custom actions `on: workflow_run` (because of read-only restrictions), and I want to see some status directly on PR. By default `workflow_run` status is displayed only in the Actions tab.

**Comment by yury-s** [comment]
@chrispat our use case:

We have a bunch of workflows running across various environments, see e.g. this [tests_primary.yml](https://github.com/microsoft/playwright/blob/dd9a49690b2d93da2a068a53ea891896810caafb/.github/workflows/tests_primary.yml). Once the tests from one of the workflows finish we need to collect all their result artifacts and produce aggregated test report which is then published to the PR that triggered the checks. Since we need access to the secrets to upload the report, the report generator runs in its own workflow [create_test_report.yml](https://github.com/microsoft/playwright/blob/dd9a49690b2d93da2a068a53ea891896810caafb/.github/workflows/create_test_report.yml) which is triggered by `workflow_run`. We currently publish the results to the pull request via `checks.create` rest api call with `head_sha` from the pull request. It all works well except for the check being put arbitrarily into one of the unrelated github actions worfklows, see e.g. [this run](https://github.com/microsoft/playwright/pull/23594/checks?check_run_id=14109915144) where the check for some reason attributed to `electron` job:

<img width="419" alt="image" src="https://github.com/community/community/assets/9798949/f7aeae6e-a3b5-4c02-8f4c-37cc5a5197b8">

We need this check to either be shown at the top level next to the workflow runs or at least a way to specify which workflow_run it should be displayed under.

**Comment by czpython** [comment]
@yury-s 
Hello, did you solve this somehow?

**Comment by yoanm** [comment]
👋 

Hitting the issue too, any advice GitHub ? 
_More or less the same use case than [there](https://github.com/orgs/community/discussions/14891#discussioncomment-6137735), `workflow_run` workflow requiring to add a check run on triggering workflow, but currently the check run is attached to any workflow executed on the run (rather than the one which triggered the `workflow_run`)_

Ability to specify the check suite would be a nice workaround 👍

**Comment by nagarrianharshit** [comment]
Facing the same issue, 
My use-case: I am creating a check run on my PR commit, since I am triggering the workflow on issue_comment, the check runs are not automatically created, So I wanted to create one and associate it to my workflow, I am able to create it but I think it is not associated to any workflow.

**Comment by thompsonjared** [comment]
Running into the same issue, would really appreciate GitHub supporting this use case. It's a clear value add than a bunch of developers are trying to take advantage but running into the same issue.

The idea that check runs are meant for apps besides GitHub actions (if that's even the case, it's hard to tell) seems like such an arbitrary miss.

**Comment by lewis-jackson-bots** [comment]
This has been tripping us up too, I want to post either `queued` or `successful` (skipped) checks for some tests that we need to have checks for due to a weird configuration with the merge queue. But since we run many jobs per commit on a PR, the checks come through with random names. They should come through as something like `E2E Tests / Shard 1`, but can come through as `Linting / Shard 1`, or `Static Analysis / Shard 1`.
